core/src/main/java/it/pagopa/selfcare/onboarding/core/InstitutionServiceImpl.java

```diff
@@ -118,13 +118,25 @@ public void onboardingPaAggregator(OnboardingData onboardingData) {
     }
 
     @Override
-    public void onboardingCompanyV2(OnboardingData onboardingData) {
+    public void onboardingCompanyV2(OnboardingData onboardingData, String userFiscalCode) {
         log.trace("onboardingProductAsync start");
         log.debug("onboardingProductAsync onboardingData = {}", onboardingData);
+        InstitutionInfoIC userBusinesses = partyRegistryProxyConnector.getInstitutionsByUserFiscalCode(userFiscalCode);
+        if(businessIsNotRelatedToUser(userBusinesses, onboardingData.getTaxCode())){
```

Comment:
You could Improve logging to provide more context, especially when exceptions are thrown:
 
```
 log.error("User with fiscal code {} is not authorized to onboard business with tax code {}", 
              userFiscalCode, onboardingData.getTaxCode());
```

Reply:
Can we log the user fiscal code or should we keep it out of the log?

Reply:
you could log only first 3 characters and the padding with ****...for ex. `RFL****`